src/components/GoogleAddressLookup/component.js

```diff
@@ -0,0 +1,297 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import withReduxForm from '../../libs/hocs/withReduxForm';
+import Lookup from '../Lookup';
+import LocationIcon from './icons/locationIcon';
+import PoweredByGoogleLogo from './icons/poweredByGoogle';
+import { uniqueId } from '../../libs/utils';
+import getSuggestions from './helpers/getSuggestions';
+import getSearchForOption from './helpers/getSearchForOption';
+import getFormattedValue from './helpers/getFormattedValue';
+import SelectedLocationIcon from './icons/selectedLocationIcon';
+import './styles.css';
+
+class PlacesLookupComponent extends Component {
+    constructor(props) {
+        super(props);
+        this.lookupId = uniqueId('gaddrlookup-input');
+        this.initialized = false;
+        this.handleChange = this.handleChange.bind(this);
+        this.handleSearch = this.handleSearch.bind(this);
+
+        this.state = {
+            initialized: false,
+            isSearching: false,
+            places: [],
+            suggestions: [],
+        };
+    }
+
+    componentDidUpdate({ isScriptLoaded: prevIsScriptLoaded }) {
+        const { isScriptLoaded, isScriptLoadSucceed } = this.props;
+        if (!prevIsScriptLoaded && isScriptLoaded && isScriptLoadSucceed) {
+            this.initComponent();
+        }
+    }
+
+    getContainerClassName() {
+        const { className } = this.props;
+        return classnames('rainbow-google-address-lookup_container', className);
+    }
+
+    initComponent() {
+        this.autocompleteService = new window.google.maps.places.AutocompleteService();
+        this.initialized = true;
+    }
+
+    handleChange(option) {
+        const { onChange } = this.props;
+
+        if (!option) {
+            return onChange(null);
+        }
+
+        const { places } = this.state;
+        let value = places.find(place => place.place_id === option.id);
+        if (!value) {
+            value = option.data;
+        }
+
+        return onChange(value);
+    }
+
+    handleSearch(value) {
+        if (!this.initialized) return;
+
+        if (value && this.state.searchValue !== value) {
+            this.searchGooglePlaces(value)
+                .then(results => this.processSearchResults(results, value))
+                .catch(this.handleError(value));
```

Comment:
use in this way:
```.catch(() => this.handleError(value));```
because here you need to pass a callback that will be called later and here you are calling it right away